presto-main/src/main/java/com/facebook/presto/memory/QueryContext.java

```diff
@@ -278,6 +290,10 @@ public synchronized void setMemoryPool(MemoryPool newMemoryPool)
         }
         ListenableFuture<?> future = memoryPool.moveQuery(queryId, newMemoryPool);
         memoryPool = newMemoryPool;
+        if (resourceOverCommit) {
+            // Reset the memory limits based on the new pool assignment
+            setResourceOvercommit();
```

Comment:
This statement looks quite weird.  if resource `resourceOverCommit` is true, then `setResourceOvercommit? -- looks unnecessary? 

My understanding is we call `setResourceOvercommit()` to reset memory limits based on the pool assignment. If that's the case maybe elaborate a bit in the comment? e.g. 
```
While resourceOverCommit is already set to true, we still call setResourceOvercommit() to 
reset the memory limits based on the new pool assignment
```

Reply:
Alternatively, you can just do what is done in the PrestoSQL repo right? https://github.com/prestosql/presto/pull/6349/files#diff-f218331177a9762b996c0276315db720a7b987ff7a0455aca45993db22a06b15R272 :)

Reply:
Actually, since it's confirmed by PrestoSQL side PR. I will just merge this :) . Leave minor comment tweak/ refactor as future work :)